pkg/ipmi/mock.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipmi
+
+import (
+	"os"
+	"syscall"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+// mock is for testing purposes only
+type mock struct {
+	unixSyscall func(uintptr, uintptr, uintptr, uintptr) (uintptr, uintptr, unix.Errno)
+
+	// fileSyscallConn gives the option to overwrite the actual function call with a dummy function when writing tests.
+	fileSyscallConn func(f *os.File) (syscall.RawConn, error)
+
+	// fileSetReadDeadline gives the option to overwrite the actual function call with a dummy function when writing tests.
+	fileSetReadDeadline func(f *os.File, t time.Duration) error
+
+	// connRead gives the option to overwrite the actual function call with a dummy function when writing tests.
+	connRead func(f func(fd uintptr) bool, conn syscall.RawConn) error
+}
+
+func (m *mock) SendRequest(req *request) error {
+	return nil
+}
+
+func (m *mock) ReceiveResponse(msgID int64, resp *response, buf []byte) ([]byte, error) {
+	return buf, nil
+}
+
+func (m *mock) GetFile() *os.File {
+	return nil
+}
+
+func (m *mock) Close() error {
+	return nil
+}
+
+func getMockIPMI() *IPMI {
```

Comment:
i am confused about how this mock is working... the mock functions don't do anything

I also expected the mock to be a real &dev{} with testsyscalls{} rather than having its own implementations of SendRequest and ReceiveResponse?

Reply:
Ok, what shall I do?

Reply:
Now that we have figured out how to run the tests in the correct CI job with the correct configuration I can scrap the mock and throw out the tests, because they wont be run anyway.

Reply:
I don't know what to do here. Should I remove it, try to rework it, leave it?

I can remember that trying to mock the file gave me headache because the usage of file descriptor and rawconn. I couldnt figure out a way to mock the behavior properly, so I decoupled it completly form the underlaying functions.